src/vivarium/framework/components/parser.py

```diff
@@ -87,23 +88,48 @@ def get_components(
 
         """
         if isinstance(component_config, ConfigTree):
-            component_list = self.parse_component_config(component_config.to_dict())
+            component_list = self.parse_component_config(component_config)
         else:  # Components were specified in a list rather than a tree.
-            component_list = component_config
-        component_list = self.prep_components(component_list)
-        return self.import_and_instantiate_components(component_list)
+            component_list = [
+                self.create_component_from_string(component) for component in component_config
+            ]
+        return component_list
+
+    def parse_component_config(self, component_config: ConfigTree) -> List[Component]:
+        """
+        Helper function for parsing a ConfigTree into a flat list of Components.
+
+        This function converts the ConfigTree into a dictionary and passes it
+        along with an empty prefix list to
+        :meth:`process_level <ComponentConfigurationParser.process_level>`. The
+        result is a flat list of components.
 
-    def parse_component_config(
-        self, component_config: Dict[str, Union[Dict, List]]
-    ) -> List[str]:
+        Parameters
+        ----------
+        component_config
+            A ConfigTree representing a hierarchical component specification blob.
+
+        Returns
+        -------
+        List[Component]
+            A flat list of Components
+        """
+        if not component_config:
+            return []
```

Comment:
When would component_config ever be empty?

Reply:
I believe it would be something like:
```yaml
components:

```

which would create a simulation with no components. I'm not sure if that's a desirable input, but in any case, this is not a change from the way it previously worked (see `parse_component_config_to_list` in the previous version).